### PR TITLE
Reserved keywords when using enums

### PR DESCRIPTION
**Pull request by danf0rth**
Fixing https://github.com/facebookincubator/ent/issues/275

Probably the best solution what i found to keep names and database entities as they are and fix arguments or types with suffix.

### CODE REVIEW

File: entc/gen/template/meta.tmpl [hunk 1]
```diff
@@ -124,13 +123,14 @@ var (
 		}
 
 		{{ $name := $f.Validator -}}
-		// {{ $name }} is a validator for the "{{ $f.Name }}" field enum values. It is called by the builders before save.
-		func {{ $name }}({{ $f.Name }} {{ $enum }}) error {
-			switch {{ $f.Name }} {
+		{{ $arg := sanitize $f.Name }}
```
Comment (a8m) [inline entc/gen/template/meta.tmpl]:
```suggestion
		{{ $arg := receiver $f.Name }}
```

Use the existing `receiver` function. It should cover these cases.

**Comment by danf0rth** [comment]
@a8m btw, `receiver` will not helps when someone specify "go" in enum name (and maybe some other keywords also).

**Comment by a8m** [comment]
Oh, I see. But other validators use the `BuilderField` function. See: [meta.tmpl](https://github.com/facebookincubator/ent/blob/master/entc/gen/template/meta.tmpl#L80)

**Comment by danf0rth** [comment]
It seems that this is the best solution, so i removed unused function that i added in my first commit. Hope is it ok now :) Thanks for your advices.

File: entc/gen/template/meta.tmpl [hunk 2]
```diff
@@ -100,7 +100,6 @@ var (
 		{{ $enum := trimPackage $f.Type.String $.Package }}
 		// {{ $enum }} defines the type for the {{ $f.Name }} enum field.
 		type {{ $enum }} string
-
```
Comment (a8m) [inline entc/gen/template/meta.tmpl]: Can you add back please this line? Thanks

Comment (danf0rth) [reply entc/gen/template/meta.tmpl]: Added back :)

**Pull request merged**